lib/assets/javascripts/cartodb/old_common/forms/filters_combo.js

```diff
@@ -16,7 +16,13 @@
     },
 
     _formatResult: function(data) {
-      return  '<span class="value">' + data.id + '</span>' + '<span class="type">' + (data.text && data.text.charAt(0)) + '</span>'
+      // Titlecase column name if synced table
+      var name = data.id.slice();
+      if (table.table.get('synchronization') && table.table.get('synchronization').from_external_source){
+        name = Sugar.String.titleize(name);
+      }
+
+      return  '<span class="value">' + name + '</span>' + '<span class="type">' + (data.text && data.text.charAt(0)) + '</span>'
```

Comment:
Why test `charAt(0)` here? Does that return something different than `data.text.length`?